### PR TITLE
RHOAIENG-4533: Add images based on Python 3.11

### PR DESCRIPTION
**Pull request by caponetto**
https://issues.redhat.com/browse/RHOAIENG-4533

## Description

Add images based on Python 3.11

## How Has This Been Tested?

Notebook tests plus manual sanity checks on some images. We can handle issues separately if we find them afterward.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: Makefile
Comment (jiridanek) [inline Makefile:578]: what's the semicolon good for? just asking

**Comment by jiridanek** [comment]
/lgtm

these addition are not breaking the 2024a and 2023b images, so it should be harmless to add this to the branch (after the manual sanity checks are done)

File: runtimes/rocm-pytorch/ubi9-python-3.11/Pipfile
Comment (jstourac) [inline runtimes/rocm-pytorch/ubi9-python-3.11/Pipfile:7]: Should we want to have applied changes in #646 and #656 here already or we plan to perform these changes in the future? :thinking:

Comment (caponetto) [reply runtimes/rocm-pytorch/ubi9-python-3.11/Pipfile:7]: Yep. I'm watching what needs to be incorporated into this PR. 
#646 is already in. #656 will be in soon.

File: Makefile
Comment (caponetto) [reply Makefile:578]: I was out of habit since some lines have it while others don't.

File: runtimes/rocm-pytorch/ubi9-python-3.11/Pipfile
Comment (caponetto) [reply runtimes/rocm-pytorch/ubi9-python-3.11/Pipfile]: #656 done in e98ca27bf616397bcd319a5059b9705951d51852

File: base/ubi9-python-3.11/Dockerfile
Comment (CastawayEGR) [inline base/ubi9-python-3.11/Dockerfile:11]: I build a test build of some of the images and noticed that we may want to update this to reflect the _python-3.11_ instead of _python-3.9_ in the labels here. My apologies if you are already aware.

Cheers,
Tipton

Comment (caponetto) [reply base/ubi9-python-3.11/Dockerfile]: Thanks! Good catch! Fixed in 7ba66c703d293c4b57572b8621c0d1ac1f512f80

**Comment by caponetto** [comment]
https://github.com/opendatahub-io/notebooks/pull/635 will be applied to this PR shortly.

File: jupyter/datascience/ubi9-python-3.11/kustomize/base/kustomization.yaml
Comment (harshad16) [inline jupyter/datascience/ubi9-python-3.11/kustomize/base/kustomization.yaml:6]:
```suggestion
namePrefix: jupyter-datascience-ubi9-python-3-11-
commonLabels:
  app: jupyter-datascience-ubi9-python-3-11
```

File: jupyter/intel/ml/ubi9-python-3.11/kustomize/base/kustomization.yaml
Comment (harshad16) [inline jupyter/intel/ml/ubi9-python-3.11/kustomize/base/kustomization.yaml:6]:
```suggestion
namePrefix: jupyter-intel-ml-ubi9-python-3-11-
commonLabels:
  app: jupyter-intel-ml-ubi9-python-3-11
```

File: jupyter/intel/pytorch/ubi9-python-3.11/kustomize/base/kustomization.yaml
Comment (harshad16) [inline jupyter/intel/pytorch/ubi9-python-3.11/kustomize/base/kustomization.yaml:6]:
```suggestion
namePrefix: jupyter-intel-pytorch-ubi9-python-3-11-
commonLabels:
  app: jupyter-intel-pytorch-ubi9-python-3-11
```

File: jupyter/intel/tensorflow/ubi9-python-3.11/kustomize/base/kustomization.yaml
Comment (harshad16) [inline jupyter/intel/tensorflow/ubi9-python-3.11/kustomize/base/kustomization.yaml:6]:
```suggestion
namePrefix: jupyter-intel-tensorflow-ubi9-python-3-11-
commonLabels:
  app: jupyter-intel-tensorflow-ubi9-python-3-11
```

File: jupyter/minimal/ubi9-python-3.11/kustomize/base/kustomization.yaml
Comment (harshad16) [inline jupyter/minimal/ubi9-python-3.11/kustomize/base/kustomization.yaml:6]:
```suggestion
namePrefix: jupyter-minimal-ubi9-python-3-11-
commonLabels:
  app: jupyter-minimal-ubi9-python-3-11
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
Great work, 
some minor changes

**Comment by caponetto** [comment]
Thanks @harshad16! All in in https://github.com/opendatahub-io/notebooks/pull/659/commits/87c6b1d73fda69912ef2d2fe6a55cd73b4821eea

File: .github/workflows/pr-merge-image-delete.yml
Comment (jstourac) [inline .github/workflows/pr-merge-image-delete.yml:113]: Technically this set of fixes would deserve to be in a separate PR. But probably not necessary to hassle with this.

**Comment by caponetto** [comment]
/hold 
I'm double-checking some things

**Comment by caponetto** [comment]
https://github.com/opendatahub-io/notebooks/pull/667 will be applied to this PR shortly.

**Comment by caponetto** [comment]
@harshad16 any other request from your side?

**Comment by atheo89** [comment]
/lgtm
However, I recommend merging this PR into a feature branch instead of the main branch. 
The main branch is currently in sync with the 2024a release, however these changes (python 3.11) are intended for the upcoming 2024b release. Merging directly into the main branch could disrupt the current workflow. What do you think? 
@harshad16

**Comment by caponetto** [comment]
@atheo89 We had that discussion last week and the conclusion is that we can merge this code to `main` as long as we don't touch the manifests. Do you see any other issues with this approach?

**Comment by atheo89** [comment]
Thank you for the heads up! These changes don't affect the 2024a builds, so we are good! 
/lgtm

**Comment by harshad16** [comment]
>Merging directly into the main branch could disrupt the current workflow. What do you think?

as shared, this discussed and the conclusion was to keep them in main for consistent development.

Thank you for the work 

/lgtm
/approve

**Pull request merged**
